rpcs3/Emu/Cell/lv2/sys_time.cpp

```diff
@@ -131,25 +131,20 @@ u64 get_timebased_time()
 // Returns some relative time in microseconds, don't change this fact
 u64 get_system_time()
 {
-	while (true)
-	{
 #ifdef _WIN32
-		LARGE_INTEGER count;
-		verify(HERE), QueryPerformanceCounter(&count);
-
-		const u64 time = count.QuadPart;
-		const u64 freq = s_time_aux_info.perf_freq;
-
-		const u64 result = time / freq * 1000000u + (time % freq) * 1000000u / freq;
+	// Pull the time directly from Windows shared page (Constant location on all Windows machines)
+	return *reinterpret_cast<u64*>(0x7ffe0014) / 10;
```

Comment:
What the hell is this? It should have very low resolution. And RDTSC isn't slow.

Reply:
I agree, RDTSC is quite possibly the best way to get low resolution time on windows and this will not be run often. Surely this change won't be much quicker at all, and if this location ever does change..... Reeks of low level unsafe hack to me.

20% of your code is run 80% of the time. We should focus on the 20%, not the 0.01%.

Reply:
@Nekotekina 100ns resolution, and it updates whenever a clock interrupt happens, so its very accurate. Haven't though of RDTSC though, shared page was what came up to mind, but you're right.
@meladath There is nothing unsafe or hacky about it - it is a documented and future proof feature, added in Windows for this specific reason. Also, I'm not optimizing for no reason. This is the only instance of time measurement I changed, and for a reason:
![performance](https://user-images.githubusercontent.com/318789/35485572-589b6e4a-046a-11e8-9e34-c3e0fe5179bf.png)

Reply:
I stand to be corrected. How is it with your code?
Also, do you have a link to this documentation? I had a quick google but failed to find it.

Reply:
@meladath https://www.geoffchappell.com/studies/windows/km/ntoskrnl/structs/kuser_shared_data.htm

Reply:
There is no way it'll update every 100ns (or at least 1000ns). So it can't have 100ns resulution. The value uses 100ns units but it's not the resolution.